buddy.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <cstddef>
+#include <stdexcept>
+
+#include "buddy.h"
+
+class BuddyAllocator {
+    private:
+        struct buddy_alloc allocator;
+        struct page_info info; // I didn't understand the point of the pages though
+    public:
+        BuddyAllocator(std::byte* memory, std::size_t size) {
+            buddy_startup(reinterpret_cast<char*>(memory), size, &info, 1);
```

Comment:
I think you forgot to assign to `allocator` here?

Reply:
yes